### DISCUSSION TITLE
how can I customzie the error message type?

### DISCUSSION THREAD

**Discussion by yaman3bd** [react-hook-form/react-hook-form#9619] [Q&A]
I am using yup for form validation, and i18next 
so I have to customize the message to be an object instead of a string
```ts
setLocale({
  mixed: {
    default: { key: "validation.field_invalid" },
    required: { key: "validation.field_required" }
  },
  array: {},
  string: {
    min: ({ min }) => ({ key: "validation.field_min_length", values: { min } })
  },
  number: {
    min: ({ min }) => ({ key: "validation.field_number_min_length", values: { min } }),
    max: ({ max }) => ({ key: "validation.field_number_max_length", values: { max } })
  }
});
```
but when I want to display the error key I get this errors:
`TS2339: Property 'key' does not exist on type 'string'.`
`TS2339: Property 'values' does not exist on type 'string'.`
```ts
errors={t(errors.title?.message?.key ?? "", { min: errors.title?.message.values.min })}
```
is there anyway that I can change the error message type from string to whatever I want?
to be an object like:
```ts
message: { key: string, values?: any }
```

**Comment by Moshyfawn** [comment]
No, the error message is a `string` by design

**Reply by yaman3bd** [reply]
I think there should be a way by making a custom react-hook-form.d.ts
I tried it, it worked but I lost all the original types, so I have to do all the types my self

**Reply by Moshyfawn** [reply]
Even if providing custom RHF type defs worked in some case for you, RHF is expecting a `string` for the error message value **by design** and **does not support** any other message types like `object` or `Element`. Overriding default RHF types and hitting an issue with error messages is not considered a bug

**Reply by yaman3bd** [reply]
I am using `yup` for the validation and when I customize the validation message in `yup`, RHF returns whatever the message

**Comment by yaman3bd** [comment]
@bluebill1049 do you have any idea about this?
